src/celeritas/decay/interactor/MuDecayInteractor.hh

```diff
@@ -0,0 +1,242 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/decay/interactor/MuDecayInteractor.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Constants.hh"
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "corecel/data/StackAllocator.hh"
+#include "corecel/math/Algorithms.hh"
+#include "celeritas/Quantities.hh"
+#include "celeritas/decay/data/MuDecayData.hh"
+#include "celeritas/phys/FourVector.hh"
+#include "celeritas/phys/Interaction.hh"
+#include "celeritas/phys/ParticleTrackView.hh"
+#include "celeritas/phys/Secondary.hh"
+#include "celeritas/random/distribution/IsotropicDistribution.hh"
+#include "celeritas/random/distribution/RejectionSampler.hh"
+#include "celeritas/random/distribution/UniformRealDistribution.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Perform muon decay.
+ *
+ * Only one decay channel is implemented, with muons decaying to
+ * \f[
+ * \mu^- \longrightarrow e^- \overline{\nu}_e \nu_\mu
+ * \f]
+ * or
+ * \f[
+ * \mu^+ \longrightarrow e^+ \nu_e \overline{\nu}_\mu
+ * \f].
+ *
+ * This interactor follows \c G4MuonDecayChannel and the Physics Reference
+ * Manual, Release 11.2, section 4.2.3.
+ *
+ * \warning Neutrinos are currently not returned in this interactor as they
```

Comment:
I think this should be a `note` ... no one is going to be doing EM physics and neutrino physics inside the same calculation since the cross sections are so absurdly different.

Reply:
They won't be doing neutrino physics per se, but vertex quantities won't be conserved, and reconstruction/filtering for analyses might depend directly on verification of conservation of E and p to reconstruct a mother particle from an observed signal. So I thought it was reasonable to have a warning.

Reply:
Geant4 just returns all three particles. So if the user doesn't want the neutrinos to be transported and/or use their initial state, then yes, the user would likely have that coded in their `G4VUserTrackingAction` implementation.